Pocket/pocket-ios
# Add Tags Feature Branch

## Summary
Users can add tags to a saved item by using existing tags or inputing new tags via text field

## References 
IN-227, IN-231, IN-232

## Implementation Details
See previous PRs for implementation details:
* https://github.com/Pocket/pocket-ios/pull/223
* https://github.com/Pocket/pocket-ios/pull/227
* https://github.com/Pocket/pocket-ios/pull/232

## Test Steps
- [ ] Given a user has navigated to the Reader View for a Saved Item,
and the user taps on the overflow menu button on the top corner, 
then the user should see an Add Tags option.

- [x] Given a user has navigated to their Saves,
and the user taps on the overflow menu button, 
then the user should see an Add Tags option.

- [x] Given a user has navigated to their Archive list,
and the user taps on the overflow menu button, 
then the user should see an Add Tags option.

- [x] Given a user has tapped on the Add Tags option for a Saved Item,
and the user views the text field (which is attached to the keyboard),
then the user should be able to enter and add tags.

- [x] Given a user has tapped on the Add Tags option for a Saved Item,
and the user inputs a tag into the text field, 
then the user should be able save it.

- [x] Given a user is on the Add Tags screen, 
and the user views the all tags section, 
then the user should be able to see existing tags (unless included in the text field).

- [x] Given a user is on the Add Tags screen, 
and the user taps on one of the existing tags, 
then the user should see it appear in the text field.

- [x] Given a user has tapped on the Add Tags option for a Saved Item,
and the user taps on a tag, 
then the user should be able to delete it. 

Note: Changes should also reflect server side and reflect on web.

## PR Checklist:

## Screenshots
https://user-images.githubusercontent.com/6743397/187524957-5751758d-e932-4fb8-be8b-4a283ce59c74.mp4
